openedx/core/operations.py

```diff
@@ -1,14 +1,32 @@
 import os
 import signal
 import tempfile
+import gc
 
 from datetime import datetime
 from meliae import scanner
 
 
 def dump_memory(signum, frame):
-    """Dump memory stats for the current process to a temp directory. Uses the meliae output format."""
-    scanner.dump_all_objects('{}/meliae.{}.{}.dump'.format(tempfile.gettempdir(), datetime.now().isoformat(), os.getpid()))
+    """
+    Dump memory stats for the current process to a temp directory.
+    Uses the meliae output format.
+    """
+
+    timestamp = datetime.now().isoformat()
+
+    scanner.dump_all_objects('{}/meliae.{}.{}.pre-gc.dump'.format(
+        tempfile.gettempdir(), timestamp, os.getpid()
+    )
+
+    # force garbarge collection
+    for gen in xrange(3):
+        gc.collect(gen)
+        scanner.dump_all_objects(
+            '{}/meliae.{}.{}.gc-gen-{}.dump'.format(
+                tempfile.gettempdir(), timestamp, os.getpid(), gen
+            )
+        )
```

Comment:
Optional: would it be cleaner to generate the static part of the format string first, and then just plug in the generation at the end?

So, something like `format_str = '{}/meliae.{}.{}.{{}}.dump'.format(tempfile.gettempdir(), datetime.now().isoformat(), os.getpid()` and then `format_str.format('pre-gc')`?